rpcserver.go

```diff
@@ -2447,6 +2447,67 @@ func (s *rpcServer) ExpectSidecarChannel(ctx context.Context,
 	return &poolrpc.ExpectSidecarChannelResponse{}, nil
 }
 
+// DecodeSidecarTicket decodes the base58 encoded sidecar ticket into its
+// individual data fields for a more human-readable representation.
+func (s *rpcServer) DecodeSidecarTicket(ctx context.Context,
+	req *poolrpc.SidecarTicket) (*poolrpc.DecodedSidecarTicket, error) {
+
+	// Parse the ticket from its string encoded representation.
+	t, err := sidecar.DecodeString(req.Ticket)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding ticket: %v", err)
+	}
+
+	serializePubKey := func(key *btcec.PublicKey) []byte {
+		if key == nil {
+			return nil
+		}
+
+		return key.SerializeCompressed()
+	}
+
+	resp := &poolrpc.DecodedSidecarTicket{
+		Id:                        t.ID[:],
+		Version:                   uint32(t.Version),
+		State:                     t.State.String(),
+		OfferCapacity:             uint64(t.Offer.Capacity),
+		OfferPushAmount:           uint64(t.Offer.PushAmt),
+		OfferLeaseDurationBlocks:  t.Offer.LeaseDurationBlocks,
+		OfferSignPubkey:           serializePubKey(t.Offer.SignPubKey),
+		OfferAuto:                 t.Offer.Auto,
+		ExecutionPendingChannelId: nil,
+	}
+
+	if t.Offer.SigOfferDigest != nil {
+		resp.OfferSignature = t.Offer.SigOfferDigest.Serialize()
+	}
+
+	if t.Recipient != nil {
+		resp.RecipientMultisigPubkeyIndex = t.Recipient.MultiSigKeyIndex
+
+		resp.RecipientNodePubkey = serializePubKey(
+			t.Recipient.NodePubKey,
+		)
+		resp.RecipientMultisigPubkey = serializePubKey(
+			t.Recipient.MultiSigPubKey,
+		)
+	}
+
+	if t.Order != nil {
+		resp.OrderBidNonce = t.Order.BidNonce[:]
+
+		if t.Order.SigOrderDigest != nil {
+			resp.OrderSignature = t.Order.SigOrderDigest.Serialize()
+		}
+	}
+
+	if t.Execution != nil {
+		resp.ExecutionPendingChannelId = t.Execution.PendingChannelID[:]
```

Comment:
Do we want to return raw byte arrays here and for `OrderBidNonce` and `ID`? I see in the next commit we use `printJson` but perhaps we could use a hex encoding for these byte buffers instead?

Reply:
We're using `github.com/lightninglabs/protobuf-hex-display` for the `printRespJSON()`, so it will be printed on the command line as hex. I actually prefer having things as raw bytes on the protobuf level. That does make it a bit more difficult to use with REST since it'll be base64 by default. But using hex encoded strings just because of that feels kind of inefficient. Not sure about this tbh. What's your take, @Roasbeef?

Reply:
Great that `protobuf-hex-display` takes care of this. As for REST I'd say base64 is probably more convenient to use than hex encoding.